pygmt/src/grdhisteq.py

```diff
@@ -0,0 +1,358 @@
+        *,
+        output_type="pandas",
+        outfile=None,
+        divisions=None,
+        quadratic=None,
+        verbose=None,
+        region=None,
+    ):
+        r"""
+        Perform histogram equalization for a grid.
+
+        Histogram equalization provides a way to highlight data that has most
+        values clustered in a small portion of the dynamic range, such as a
+        grid of flat topography with a mountain in the middle. Ordinary gray
+        shading of this grid (using :meth:`pygmt.Figure.grdimage` or
+        :meth:`pygmt.Figure.grdview`) with a linear mapping from topography to
+        graytone will result in most of the image being very dark gray, with
+        the mountain being almost white. :meth:`pygmt.grdhisteq.compute_bins`
+        can provide a list of data values that divide the data range into
+        divisions which have an equal area in the image [Default is 16 if
+        ``divisions`` is not set]. The :class:`pandas.DataFrame` or ASCII file
+        output can be used to make a colormap with :meth:`pygmt.makecpt` and an
+        image with :meth:`pygmt.Figure.grdimage` that has all levels of gray
+        occuring equally.
+
+        Full option list at :gmt-docs:`grdhisteq.html`
+
+        Parameters
+        ----------
+        grid : str or xarray.DataArray
+            The file name of the input grid or the grid loaded as a DataArray.
+        outfile : str or bool or None
+            The name of the output ASCII file to store the results of the
+            histogram equalization in.
+        output_type : str
+            Determine the format the xyz data will be returned in [Default is
+            ``pandas``]:
+
+                - ``numpy`` - :class:`numpy.ndarray`
+                - ``pandas``- :class:`pandas.DataFrame`
+                - ``file`` - ASCII file (requires ``outfile``)
+        divisions : int
+            Set the number of divisions of the data range.
+        quadratic : bool
+            Perform quadratic equalization [Default is linear].
+        {R}
+        {V}
+
+        Returns
+        -------
+        ret: pandas.DataFrame or None
+            Return type depends on the ``outfile`` parameter:
+
+            - pandas.DataFrame if ``outfile`` is True or None
+            - None if ``outfile`` is a str (file output is stored in
+              ``outfile``)
+
+        Example
+        -------
+        >>> import pygmt  # doctest: +SKIP
+        >>> # Load a grid of @earth_relief_30m data, with an x-range of 10 to
+        >>> # 30, and a y-range of 15 to 25
+        >>> grid = pygmt.datasets.load_earth_relief(
+        ...     resolution="30m", region=[10, 30, 15, 25]
+        ... )  # doctest: +SKIP
+        >>> # Find elevation intervals that divide the data range into
+        >>> # divisions which have an equal area in the original grid.
+        >>> bins = pygmt.grdhisteq.compute_bins(
+        ...     grid=grid, divisions=5
+        ... )  # doctest: +SKIP
+
```

Comment:
Just to illustrate what `compute_bins` is actually doing.
```suggestion
        >>> print(bins)  # doctest: +SKIP
              start      stop  bin_id
        0 -3.023719 -0.846985       0
        1 -0.846985 -0.257874       1
        2 -0.257874  0.248180       2
        3  0.248180  0.838059       3
        4  0.838059  3.023719       4

```

Reply:
That's a really good idea actually. Go for it!